### ISSUE TITLE
-openmp option is deprecated in icpc

### ISSUE THREAD

**Issue by srajama1** [trilinos/Trilinos#1427]
When enabling OpenMP with intel 17 I get tons of these

icpc: command line remark #10411: option '-openmp' is deprecated and will be removed in a future release. Please use the replacement option '-qopenmp'

Can we change the configure logic to handle this ?

@trilinos/framework @bartlettroscoe

**Comment by ibaned** [comment]
Going further, `-fopenmp` should be accepted by all OpenMP-supporting compilers (at least the ones I've seen including Intel).

**Comment by bartlettroscoe** [comment]
Looks like this is a CMake issue, not a TriBITS issue.  Looks like these options come from `FIND_PACKAGE(OpenMP)` as shown at:

* https://github.com/trilinos/Trilinos/blob/47c09c72b28f09bb56301a50cea9d6ccaa134164/cmake/tribits/core/package_arch/TribitsGlobalMacros.cmake#L1893

Therefore, please refer to the documentation for the FindOpenMPI.cmake module for CMake 2.8.11:

* https://cmake.org/cmake/help/v2.8.11/cmake.html#module:FindOpenMP

My advise is to try a newer version of CMake and see if that solves the problem.

**Comment by ibaned** [comment]
For what its worth, I never specify `Trilinos_ENABLE_OpenMP`, instead I just add `-fopenmp` to the `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS` explicitly. I guess thats also more reliable considering CMake's mistake.

**Comment by srajama1** [comment]
@bartlettroscoe Does it have to be FIND_PACKAGE(OpenMP) ? Can we do -fopenmp for older CMake compilers. Of course I can module load cmake. However, we will have to repeat this explanation multiple times to users. If there is an easy fix in TriBITS, let us do it.

**Comment by bartlettroscoe** [comment]
@srajama1,

> Does it have to be FIND_PACKAGE(OpenMP) ? Can we do -fopenmp for older CMake compilers. Of course I can module load cmake. However, we will have to repeat this explanation multiple times to users. If there is an easy fix in TriBITS, let us do it.

We get in trouble when we change the behavior of raw CMake features (if we choose to use them).  In fact, I have been criticized many times for cases where TriBITS deviates from raw CMake behavior.

Anyway, I did not write any of the CMake code that added OpenMP support (I don't think I was even consulted on it).  The history of who did is seen in the original Trilinos git repo  (i.e. before the GitHub filter was done) in the commits:

```
commit 4f7805bb61ceb1ca8129e2ea851e1c5ff90c35bb
Author: Nico Schlömer <nico.schloemer@gmail.com>
Date:   Wed May 7 14:46:24 2014 +0200

    INCLUDE(FindOpenMP) -> FIND_PACKAGE(OpenMP)
    
    Fixes bug #5934.

M       cmake/tribits/package_arch/TribitsGlobalMacros.cmake
```

and

```
commit 3ba86e8574a29ed81811565f384917960e42451d
Author: James M. Willenbring <jmwille@sandia.gov>
Date:   Mon Mar 15 12:11:50 2010 -0600

    Adding a cmake option for openmp called Trilinos_ENABLE_OpenMP.  Epetra
    recently started supporting an OpenMP-aware build.
    
    =============================
    Build/Test Cases Summary
    =============================
    
    Mon Mar 22 23:13:07 MDT 2010
    
    Enabled Packages: TrilinosFramework, Epetra
    Disabled Packages: Rythmos
    Enabled all Packages
    
    Build test results:
    -------------------
    0) MPI_DEBUG => passed: Trilinos/MPI_DEBUG: passed=1112,notpassed=0 (31.3037337025 min)
    1) SERIAL_RELEASE => passed: Trilinos/SERIAL_RELEASE: passed=1156,notpassed=0 (62.1580179811 min)
    
    Local commits for this build/test group:
    ----------------------------------------
    acd874c Adding a cmake option for openmp called Trilinos_ENABLE_OpenMP.  Epetra recently started supporting an OpenMP-aware build.

M       cmake/TrilinosProbeEnv.cmake
```

So @nschloe and @jwillenbring, what should be done about this OpenMP issue?  I don't have a dog in this fight.

But given that this was added long ago, if it gets changed to not use `FIND_PACKAGE(OpenMP)` this will break many people's current configure scripts.  And in case you are suggesting that we simply copy `FindOpenMP.cmake` and make it do what is suggested, consider the mess that creates for future versions of CMake (as an example, see TriBITSPub/TriBITS#148).

And again, we get attacked when we deviate from or override raw CMake behavior like this (CC: @maherou and @nmhamster and, again, consider the counter-example TriBITSPub/TriBITS#148).

We need to stop hacking TriBITS every time we find an issue like this and instead push for more aggressive updates of CMake, assuming those issues are addressed in updated CMake versions.  And if the issue is not addressed in an already-released CMake version, then we need to pay Kitware to fix those issues and then use upgraded versions of CMake.  That is what I am being pushed to do with TriBITS very strongly by senior level staff and management.  We need to reduce the scope of TriBITS to just handle the package architecture system (which is a feature that raw CMake does not have) and get out of the business of changing raw CMake behavior for features that it does have.

**Comment by nmhamster** [comment]
@bartlettroscoe @nschloe @jwillenbring - still seeing this `icpc: command line remark #10411: option '-openmp' is deprecated and will be removed in a future release. Please use the replacement option '-qopenmp'` Note that use of `-fopenmp` is not actually the correct answer here, we should test for `-qopenmp` first and then work backwards. I know this is a hassle but it doesn't half produce some seriously large build logs.
